src/main/java/com/faforever/client/task/TaskService.java

```diff
@@ -43,9 +44,10 @@
         logger.warn("Task failed", (Throwable) throwable);
       }
     });
-
-    activeTasks.add(task);
-    executorService.execute(task);
+    Platform.runLater(() -> {
```

Comment:
I would just use JavaFx.runLater here

Reply:
Whats the difference?

Reply:
ah ok I can do that

Reply:
JavaFx checks if we are on the application thread first that way we dont unnecessarily add to the platform queue